packages/integrations/source-control/src/github/index.ts

```diff
@@ -241,12 +235,12 @@ export class GitHubSourceControl implements SourceControl {
     const linkHeader = parseLinkHeader(result.headers.link) || { next: { per_page: perPage } };
 
     const pullsTotalPages = (!('last' in linkHeader)) ? page : Number(linkHeader.last?.page);
-    const pullsPerPage =  ('next' in linkHeader) ? Number(linkHeader.next?.per_page) : Number(linkHeader.prev?.per_page);
-    
+    const pullsPerPage = ('next' in linkHeader) ? Number(linkHeader.next?.per_page) : Number(linkHeader.prev?.per_page);
+
     const pagination = {
       page: firstPagePagination?.page || page,
       perPage: perPage || firstPagePagination?.perPage || pullsPerPage, // Dejan: This can break if firstPagePagination returns different perPage -> check documentation on linkHeader ???
-      totalPages: totalPages || firstPagePagination?.totalPages || pullsTotalPages,
+      totalPages: totalPages || firstPagePagination?.totalPages || pullsTotalPages, // Refactor: should recalculate totalPages here if pulls api returns different perPage
```

Comment:
This can't be fixed without a refactor which is why I did the refactor in the first PR.

We should open an issue to monkey test the search pagination strategy, then refactor if we find the issue or alternatively, move to cursor based pagination if GitHub is the issue.

Reply:
Maybe I'm wrong, I could set perPage to the search response, but I'm not sure what that would imply..